rust/connlib/tunnel/src/gateway.rs

```diff
@@ -59,7 +60,7 @@ where
         expires_at: Option<DateTime<Utc>>,
         resource: ResourceDescription<ResolvedResourceDescriptionDns>,
     ) -> Result<ConnectionAccepted> {
-        let (resource_addresses, id) = match &resource {
+        let (resource_addresses, id, filters) = match &resource {
```

Comment:
Given the repition of this `match`, can we make a function on `ResourceDescription` that de-duplicates this?